homeassistant/components/zwave_js/lock.py

```diff
@@ -161,6 +191,42 @@ async def async_unlock(self, **kwargs: Any) -> None:
         """Unlock the lock."""
         await self._set_lock_state(STATE_UNLOCKED)
 
+    async def async_get_lock_usercode(self, code_slot: int) -> ServiceResponse:
+        """Get the usercode at index X on the lock."""
+        try:
+            code_slot_obj = get_usercode(self.info.node, code_slot)
+        except BaseZwaveJSServerError as err:
+            raise HomeAssistantError(
+                f"Unable to get usercode for lock {self.entity_id} code_slot "
+                f"{code_slot}: {err}"
+            ) from err
+        LOGGER.debug(
+            "User code at slot %s on lock %s retrieved", code_slot, self.entity_id
+        )
+        usercode: str | None = code_slot_obj.get(ATTR_USERCODE)  # type: ignore[assignment]
+        return {"usercode": usercode}
```

Comment:
why don't we return just the user code?

Reply:
I agree. Makes more sense to return the code vs an object.